contracts/svm/programs/express_relay/src/lib.rs

```diff
@@ -147,6 +152,155 @@ pub mod express_relay {
             amount,
         )
     }
+
+    pub fn swap(ctx: Context<Swap>, data: SwapArgs) -> Result<()> {
+        validate_ata(
+            &ctx.accounts.trader_input_ata.key(),
```

Comment:
the trader_input token account doesn't need to be an ATA? it is possible that a trader wants to use tokens from a non-canonical token account and the program should let them?

Reply:
maybe there are some niche hiccups associated with allowing the searcher to specify any trader TA they want? not 100% sure yet, would be good to think about this before committing to one

Reply:
made it ATA for now, we can change later

Reply:
made it ATA for now, we can change later